FDK17プロジェクト/コード/03.サウンド/CSound.cs

```diff
@@ -574,27 +579,103 @@ public double db再生速度
 		private SYNCPROC _cbEndofStream;	// ストリームの終端まで再生されたときに呼び出されるコールバック
 //		private WaitCallback _cbRemoveMixerChannel;
 
+	    /// <summary>
+	    /// Gain is applied "first" to the audio data, much as in a physical or
+	    /// software mixer. Later steps in the flow of audio apply "channel" level
+	    /// (e.g. AutomationLevel) and mixing group level (e.g. GroupLevel) before
+	    /// the audio is output.
+	    /// 
+	    /// This is currently used for mixing in the SONGVOL value, when available.
+	    /// It is also currently used for DTXViewer preview mode.
+	    ///
+	    /// It will in the near future also be used for adjustments based on
+	    /// loudness metadata, either when SONGVOL is unavailable or when loudness
+	    /// metadata is prioritized over it.
+	    /// </summary>
+	    public int Gain
```

Comment:
The new CSound properties that are all mixed together to produce the final volume level for each sound are here and below. Their doc comments are good reading.